src/db.cr

```diff
@@ -0,0 +1,7 @@
+class DB
+  @calls = Hash(String, Call).new
+
+  def save_call(call : Call) : Call
```

Comment:
It's odd that the Calls are created outside the db. How you ensure they are created in the right state.